cli/pkg/version/version.go

```diff
@@ -0,0 +1,81 @@
+package version
+
+import (
+	"fmt"
+	"regexp"
+	"strconv"
+)
+
+type Version struct {
+	Major       int
+	Patch       int
+	Incremental int
+}
+
+func Parse(ver string) (*Version, error) {
+	re := regexp.MustCompile(`(\d+)\.(\d+)(?:-\w+(\d+))?`)
+	match := re.FindStringSubmatch(ver)
+	if len(match) < 3 || len(match) > 4 {
+		return nil, fmt.Errorf("invalid format expected <major>.<patch>(-<suffix><incremental>)? %s,", ver)
+	}
+	major, err := strconv.Atoi(match[1])
+	if err != nil {
+		return nil, fmt.Errorf("error parsing major int: %s, err %w", match[1], err)
+	}
+	patch, err := strconv.Atoi(match[2])
+	if err != nil {
+		return nil, fmt.Errorf("error parsing patch int: %s, err %w", match[2], err)
+	}
+
+	if match[3] == "" {
```

Comment:
maybe change to check to len(match) == 3?

Reply:
it will always will be 4